R/functions.R

```diff
@@ -666,24 +671,36 @@ df2vw <- function(data, file_path, namespaces = NULL, keep_space = NULL,
     specChar      <- "\\(|\\)|\\||\\:|'"
     specCharSpace <- "\\(|\\)|\\||\\:| |'"
 
-    parsingNames <- function(x) {
+    parsingNames <- function(x, keepSpace=F) {
         ret <- c()
-        for (el in x)
-            ret <- append(ret, gsub(specCharSpace,'_', el))
+        if(keepSpace){
+            for (el in x)
+                ret <- append(ret, gsub(specChar,'_', el))
+        } else {
+            for (el in x)
+                ret <- append(ret, gsub(specCharSpace,'_', el))
+        }
         ret
     }
 
     # parse categorical variables
     parsingVar <- function(x, keepSpace) {
-        # remove leading and trailing spaces, then remove special characters
-        # then remove isolated underscores.
-        if (!keepSpace)
+        if (!keepSpace) # Replace special characters
             spch <- specCharSpace
-        else
+        else # Replace special characters, but keep spaces
             spch <- specChar
         gsub(spch, '_', x)
     }
-
+    # parsingVar <- function(x, keepSpace, fixed) {
```

Comment:
Any reason for adding in commented out code?

Reply:
Ah well I do that too all the time.

Reply:
I thought I would return to this variant, but there's actually no point keeping it anymore.
So I will just delete all these lines.